ios/UARCTModule/UARCTEventEmitter.m

```diff
@@ -36,37 +40,50 @@ - (instancetype)init {
     self = [super init];
     if (self) {
         self.pendingEvents = [NSMutableArray array];
+        self.knownListeners = [NSMutableSet set];
     }
 
     return self;
 }
 
-- (BOOL)sendEventWithName:(NSString *)eventName body:(id)body {
-    if (self.bridge && self.isObserving) {
+- (void)sendEventWithName:(NSString *)eventName body:(id)body {
+    if (self.bridge && self.isObserving && [self.knownListeners containsObject:eventName]) {
         [self.bridge enqueueJSCall:@"RCTDeviceEventEmitter"
                             method:@"emit"
                               args:body ? @[eventName, body] : @[eventName]
                         completion:NULL];
 
-        return YES;
+    } else {
+        @synchronized(self.pendingEvents) {
+            [self.pendingEvents addObject:@{ UARCTEventNameKey: eventName, UARCTEventBodyKey: body}];
+        }
     }
-
-    return NO;
 }
 
 - (void)addListener:(NSString *)eventName {
     self.listenerCount++;
```

Comment:
I have to admit I'm a little confused by the synchronization going on here, but that's probably because it's not clear to me what kind of multithreading environment this stuff is living in. But for instance, take our methods for adding and removing listeners. If it's necessary to synchronize on knownListeners in those methods, presumably because they can be called from different threads, should mutating self.listenerCount also be synchronized?